src/Eloquent/HybridRelations.php

```diff
@@ -272,6 +272,10 @@ public function belongsToMany(
 
         $instance = new $related;
 
+        if ($relatedPivotKey === $relation) {
+            throw new \LogicException(sprintf('In %s::%s(), the key cannot be identical to the relation name "%s". The default key is "%s".', static::class, $relation, $relation, $instance->getForeignKey().'s'));
```

Comment:
You can optionally use argument numbering to remove the duplicate `$relation` argument.

 ```suggestion
            throw new \LogicException(sprintf('In %1$s::%2$s(), the key cannot be identical to the relation name "%2$s". The default key is "%3$s".', static::class, $relation, $instance->getForeignKey().'s'));
```